src/backends/pyplot.jl

```diff
@@ -1384,10 +1392,14 @@ function _update_plot_object(plt::Plot{PyPlotBackend})
         # set the cbar position if there is one
         if haskey(sp.attr, :cbar_ax)
             cbw = sp.attr[:cbar_width]
-            # this is the bounding box of just the colors of the colorbar (not labels)
-            ex = sp[:zaxis][:extrema]
-            has_toplabel = !(1e-7 < max(abs(ex.emax), abs(ex.emin)) < 1e7)
-            cb_bbox = BoundingBox(right(sp.bbox)-cbw+1mm, top(sp.bbox) +  (has_toplabel ? 4mm : 2mm), _cbar_width-1mm, height(sp.bbox) - (has_toplabel ? 6mm : 4mm))
+
+            # Colorbar should be placed right to the axis with same vertical size
+            # we can extract bbox from axis object to make it match to it
+            cb_bbox = BoundingBox(right(py_bbox(ax)) + _cbar_width/2,  # Some paddding between colorbar and the ax
```

Comment:
For some reason, `py_bbox(ax)` no longer refers to the actual bounding box of the axis with increasing dpi. I would assume that `py_bbox(ax)` always shows the bounding box of the axes (axis frames)